compile.go

```diff
@@ -47,54 +47,72 @@ func Compile(ctx context.Context, q string, now time.Time, opts ...Option) (*Spe
 	}
 
 	s, _ := opentracing.StartSpanFromContext(ctx, "parse")
-	itrp := NewInterpreter()
-	itrp.SetOption(nowOption, nowFunc(now))
-	if err := Eval(itrp, q); err != nil {
+
+	functionCalls, scope, err := Eval(q, SetOption(nowOption, nowFunc(now)))
```

Comment:
nit, `functionCalls` is a bit of a misnomer, should we just call it `sideEffects`?

Reply:
Yes I meant to change that.